enterprise/cmd/executor/internal/worker/handler.go

```diff
@@ -271,17 +286,44 @@ func writeFiles(workspaceFileContentsByPath map[string][]byte, logger command.Lo
 		handle.Close()
 	}()
 
-	for path, content := range workspaceFileContentsByPath {
+	for path, wf := range workspaceFileContentsByPath {
 		// Ensure the path exists.
 		if err := os.MkdirAll(filepath.Dir(path), os.ModePerm); err != nil {
 			return err
 		}
 
-		if err := os.WriteFile(path, content, os.ModePerm); err != nil {
+		var src io.ReadCloser
+
+		// Log how long it takes to write the files
+		start := time.Now()
+		if len(wf.content) > 0 {
+			src = io.NopCloser(bytes.NewReader(wf.content))
+		} else if wf.bucket != "" && wf.key != "" {
+			src, err = store.Get(ctx, wf.bucket, wf.key)
+			if err != nil {
+				return err
+			}
+		} else {
+			return errors.Newf("no content for %s", path)
+		}
```

Comment:
I think I would do the following (bc filestore would also allow empty files to be written, and it doesn't feel necessary to forbid empty files)

```suggestion
		if wf.bucket != "" && wf.key != "" {
			src, err = store.Get(ctx, wf.bucket, wf.key)
			if err != nil {
				return err
			}
		} else {
			src = io.NopCloser(bytes.NewReader(wf.content))
		}
```